src/catalog/catalog_entry/scalar_macro_catalog_entry.cpp

```diff
@@ -0,0 +1,105 @@
+#include "duckdb/catalog/catalog_entry/scalar_macro_catalog_entry.hpp"
+#include "duckdb/catalog/catalog_entry/table_macro_catalog_entry.hpp"
+#include "duckdb/common/field_writer.hpp"
+#include "duckdb/function/scalar_macro_function.hpp"
+#include "duckdb/function/table_macro_function.hpp"
+
+namespace duckdb {
+
+MacroCatalogEntry::MacroCatalogEntry(Catalog *catalog, SchemaCatalogEntry *schema, CreateMacroInfo *info)
+    : StandardEntry(
+          (info->function->type == MacroType::SCALAR_MACRO ? CatalogType::MACRO_ENTRY : CatalogType::TABLE_MACRO_ENTRY),
+          schema, catalog, info->name),
+      function(move(info->function)) {
+	this->temporary = info->temporary;
+	this->internal = info->internal;
+}
+
+ScalarMacroCatalogEntry::ScalarMacroCatalogEntry(Catalog *catalog, SchemaCatalogEntry *schema, CreateMacroInfo *info)
+    : MacroCatalogEntry(catalog, schema, info) {
+}
+
+void ScalarMacroCatalogEntry::Serialize(Serializer &main_serializer) {
+	D_ASSERT(!internal);
+	auto &scalar_function = (ScalarMacroFunction &)*function;
+	FieldWriter writer(main_serializer);
+	writer.WriteString(schema->name);
+	writer.WriteString(name);
+	writer.WriteSerializable(*scalar_function.expression);
+	// writer.WriteSerializableList(function->parameters);
+	writer.WriteSerializableList(function->parameters);
+	writer.WriteField<uint32_t>((uint32_t)function->default_parameters.size());
+	auto &serializer = writer.GetSerializer();
+	for (auto &kv : function->default_parameters) {
+		serializer.WriteString(kv.first);
+		kv.second->Serialize(serializer);
```

Comment:
Storage of the default parameters is not tested either, could you add a test for this as well?

Reply:
Have added  new test script to explicitly check macro args see:
 test_table_macro_args.test